src/index.js

```diff
@@ -61,126 +66,197 @@ const monitorChromecastPlayer = function (player, options) {
+            // When a seek happens, cast sdk sends the `cast.framework.events.EventType.PAUSE`.
+            // We need to suppress the player.mux.emit('pause') call, because if it goes out, in our data view, seek
+            // events will become pause event
+            player.mux.emit('pause');
+          }
+          break;
+        case cast.framework.events.EventType.PLAYING:
+          if (isPaused) {
+            player.mux.emit('play');
+          }
+          if (isSeeking) {
+            isSeeking = false;
+          }
+          isPaused = false;
+          player.mux.emit('playing');
+          break;
+        case cast.framework.events.EventType.ERROR:
+          if (!options.automaticErrorTracking) { return; }
+          switch (event.detailedErrorCode) {
+            case 901:
+            case 902:
+              // specific errors to ads
+              player.mux.emit('aderror');
+              break;
+            case 903:
+            case 904:
+            // do nothing since these aren't fatal errors
+              break;
+            default:
+              player.mux.emit('error', {
+                player_error_code: event.detailedErrorCode,
+                player_error_message: event.error ? event.error.message : 'Unknown Error'
+              });
+              break;
+          }
+          break;
+        case cast.framework.events.EventType.RATE_CHANGE:
+          player.mux.emit('ratechange');
+          break;
+        case cast.framework.events.EventType.TIME_UPDATE:
+          currentTime = event.currentMediaTime;
+          player.mux.emit('timeupdate');
+          break;
+        case cast.framework.events.EventType.SEGMENT_DOWNLOADED:
+          let now = Date.now();
+          let loadData = {
+            request_event_type: 'SEGMENT_DOWNLOADED',
+            request_start: now - event.downloadTime - 1,
+            request_response_start: now - event.downloadTime,
+            request_response_end: now,
+            request_bytes_loaded: event.size,
+            request_type: 'media'
+          };
+
+          player.mux.emit('requestcompleted', loadData);
+          break;
+
+        case cast.framework.events.EventType.BREAK_STARTED:
+          inAdBreak = true;
+          player.mux.emit('adbreakstart');
+          break;
+      }
+    }
+
+    if (inAdBreak === true) {
+      // chromecast will mix player messages with ad messages
+      // seperating out these ad events and supressing unneeded messages until the adbreak is over
+      switch (event.type) {
+        case cast.framework.events.EventType.BREAK_CLIP_LOADING:
+          if (adBreakClips !== null && event.breakClipId !== undefined) {
+            let breakClip = adBreakClips.find(o => o.id === event.breakClipId);
+            let adAssetUrl = (breakClip !== undefined && breakClip.contentUrl !== undefined) ? breakClip.contentUrl : (breakClip !== undefined && breakClip.contentId !== undefined) ? breakClip.contentId : null;
```

Comment:
Okay so this can be simplified a bit:

```javascript
let breakClip = adBreakClips.find(o => o.id === event.breakClipId) || {};
let adAssetUrl = breakClip.contentUrl || breakClip.contentId;
```

That should functionally be the same thing, and I think a bit easier to reason about. Essentially, the first line finds the break clip, but if it's not found, gives us an empty object to play around with. Then the next line says give me the contentUrl, and if that doesn't exist, try to get the contentId.